velox/common/memory/Memory.h

```diff
@@ -105,6 +106,11 @@ constexpr uint16_t kDefaultAlignment = 64;
 /// its raw pointer from its parent through dropChild() and then drops the
 /// shared reference on the parent.
 ///
+/// NOTE: for the users that integrate at expression evaluation level, we don't
+/// need to build the memory pool hierarchy as described above but simply create
+/// one child memory pool from IMemoryManager::getChild() for each expression
```

Comment:
> for each expression evaluation.

Is it necessary to create new memory pool for each expression? Can we not reuse memory pool?

Reply:
It is not which depends whether user want to enforce a quota or not. Currently, ExeCtx doesn't hold a reference on the memory pool.